packages/studio-plugin/src/utils/ComponentTreeHelpers.ts

```diff
@@ -186,4 +186,44 @@ export default class ComponentTreeHelpers {
 
     return descendants;
   }
+
+  /**
+   * Returns the component UUID that is closest common parent of the two target components.
+   * This is similar to the lowest common ancestor except it cannot be one of the targets.
+   */
+  static getLowestParentUUID(
+    targetOneUUID: string | undefined,
+    targetTwoUUID: string | undefined,
+    componentTree: ComponentState[]
+  ) {
+    const LCAComponent = ComponentTreeHelpers.getLowestCommonAncestorComponent(
+      targetOneUUID,
+      targetTwoUUID,
+      componentTree
+    );
+    const TargetIsLCA =
+      targetOneUUID === LCAComponent?.uuid ||
+      targetTwoUUID === LCAComponent?.uuid;
+    return TargetIsLCA ? LCAComponent?.parentUUID : LCAComponent?.uuid;
+  }
+
+  private static getLowestCommonAncestorComponent(
+    targetOneUUID: string | undefined,
+    targetTwoUUID: string | undefined,
+    componentTree: ComponentState[]
+  ): ComponentState | undefined {
+    let LCA_UUID;
+    ComponentTreeHelpers.mapComponentTree(componentTree, (c) => {
+      if (!LCA_UUID && targetOneUUID === targetTwoUUID) {
```

Comment:
once we find the LCA, we don't need to continue mapping, right? I think it would be cleaner if we short-circuit early if LCA_UUID is defined

Reply:
is there a way to short circuit out of mapComponentTree?

Reply:
oh I just meant returning early inside this callback

Reply:
oh gotcha! added